lib/HLSL/DxilScalarizeVectorIntrinsics.cpp

```diff
@@ -27,12 +27,13 @@
 using namespace llvm;
 using namespace hlsl;
 
-static void scalarizeVectorLoad(hlsl::OP *HlslOP, const DataLayout &DL,
+static bool scalarizeVectorLoad(hlsl::OP *HlslOP, const DataLayout &DL,
                                 CallInst *CI);
-static void scalarizeVectorStore(hlsl::OP *HlslOP, const DataLayout &DL,
+static bool scalarizeVectorStore(hlsl::OP *HlslOP, const DataLayout &DL,
                                  CallInst *CI);
-static void scalarizeVectorIntrinsic(hlsl::OP *HlslOP, CallInst *CI);
-static void scalarizeVectorReduce(hlsl::OP *HlslOP, CallInst *CI);
+static bool scalarizeVectorIntrinsic(hlsl::OP *HlslOP, CallInst *CI);
```

Comment:
It seemed weird to me that all of these were updated to return bools when it looks like only scalarizeVectorWaveMatch could return false. All others are always going to return true?

Wondering if it would express intent better to revert the ones that always return true to return void. And just update the value of Changed in runOnModule.



`Changed |= scalarizeVectorLoad(HlslOP, M.getDataLayout(), CI);`

to

`scalarizeVectorLoad(HlslOP, M.getDataLayout(), CI);`
`Changed = true`

Reply:
either is fine, but the first to me felt better since we were delegating the decision of "did you change" to the function actually doing the change instead of having the caller assume the intent. 

It's not hard to imagine a future where the function that currently always does a change is updated to have an early exit. With the `bool` return route the compiler tells the editor they need to consider the updated value. Without it the editor could just edit the function and the call location won't show up in the diff so the reviewer won't know to comment "hey update the `Changed  = true`" please